--- rigid.f90.orig	2021-01-29 16:55:25.000000000 +0800
+++ rigid.f90	2021-01-29 16:54:28.000000000 +0800
@@ -393,6 +393,13 @@
   if ( diff > 1.d-6 ) write (6,'(5x,"Max |d(i,j)-d*(j,i)| = ",f9.6,/,5x, &
        & "Max |d(i,j)-d*(j,i)|/|d(i,j)|: ",f8.4,"%")') diff, difrel*100
   !
+  !  print the foce constant matrix after asr !hw
+  OPEN (unit=1000,file="IFC_sumrule.dat",status='unknown')
+  do i = 1,nat3
+    write (1000, *) (dyn2(i,j), j = 1,nat3)
+  enddo
+  !hw
+  ! 
   !  divide by the square root of masses
   !
   do na = 1,nat
